src/NzbDrone.Core/Download/Clients/DownloadStation/DownloadStation.cs

```diff
@@ -0,0 +1,323 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using FluentValidation.Results;
+using NLog;
+using NzbDrone.Common.Disk;
+using NzbDrone.Common.Http;
+using NzbDrone.Core.Configuration;
+using NzbDrone.Core.MediaFiles.TorrentInfo;
+using NzbDrone.Core.Parser.Model;
+using NzbDrone.Core.RemotePathMappings;
+using NzbDrone.Common.Extensions;
+using NzbDrone.Core.Validation;
+using System.Net;
+using static NzbDrone.Core.Download.Clients.DownloadStation.DownloadStationTorrent;
+using NzbDrone.Core.Download.Clients.DownloadStation.Proxies;
+using NzbDrone.Common.Cache;
+
+namespace NzbDrone.Core.Download.Clients.DownloadStation
+{
+    public class DownloadStation : TorrentClientBase<DownloadStationSettings>
+    {
+        protected readonly IDownloadStationProxy _proxy;
+        protected readonly ISharedFolderResolverService _sharedFolderResolver;
+
+        public DownloadStation(IDownloadStationProxy proxy,
+                               ITorrentFileInfoReader torrentFileInfoReader,
+                               IHttpClient httpClient,
+                               IConfigService configService,
+                               IDiskProvider diskProvider,
+                               IRemotePathMappingService remotePathMappingService,
+                               Logger logger,
+                               ICacheManager cacheManager,
+                               ISharedFolderResolverService sharedFolderResolver)
+            : base(torrentFileInfoReader, httpClient, configService, diskProvider, remotePathMappingService, logger)
+        {
+            _proxy = proxy;
+            _sharedFolderResolver = sharedFolderResolver;
+            _sharedFolderResolver.Settings = Settings;
```

Comment:
Can't do that, Settings is assigned after the constructor. Easiest is probably to pass it along on the GetPhysicalPath method call. 
Also, SharedFolderResolver is a singleton, while DownloadStation might not be if ppl create multiple, so it has to be stateless. Meaning the Settings.Host+Port needs to be used as an additional key (prefix maybe) in the _cache.
It's really unlikely ppl add multiple diskstations, but if they do then it goes bad.